torch/utils/_foreach_utils.py

```diff
@@ -21,6 +22,14 @@
 def _group_tensors_by_device_and_dtype(tensorlistlist: List[List[Tensor]],
                                        with_indices: Optional[bool] = False) -> \
         Dict[Tuple[torch.device, torch.dtype], List[List[Union[Tensor, int]]]]:
+
+    if is_compiling():
+        if with_indices:
+            indices = list(range(len(tensorlistlist[0])))
+            tensorlistlist.append(indices)
+
+        return {(None, None): tensorlistlist}
```

Comment:
This is the correct format, but the (None, None) will break usages like in clip_grad.py if is_compiling() since the code checks the device. 

The 100% safe alternative is to move the check outside the function call, which may feel redundant across the optimizers but can be abstracted. ccing @albanD for a second opinion.

Reply:
Oh ok, didn't realize the key was used outside the function. yeah I considered just wrapping the call. Lemme make a compile-friendly helper that does that